### PR TITLE
add getGenres support

### LINKED ISSUE #34: getGenres Support
It seems that Gonic does not support the `getGenres` endpoint. Is this in the roadmap? (I know you are busy, I just graduated and I know the final year pains.)

This is affecting a user of my Subsonic client:
https://gitlab.com/sumner/sublime-music/issues/109#note_284391742
(arguably, my client should be more resilient to not having certain endpoints missing, and that's a goal I have in the future, but I just wanted to make you aware on the server-side.)

### PR DESCRIPTION
**Pull request by Duncaen**
I never used gomigrate, so I'm not sure if this is correct.

This also adds a decoded name column to the genres table, I thought there was a indexed genre but there isn't I can remove the column if you think its not necessary.

closes #34

### CODE REVIEW

**Comment by sentriz** [comment]
hi! thank you very much for this - looks very good. 
as far as i can remember the "u dec" columns are only used for searching (so you can look up artist/tracks/albums with non latin chars with just latin char equivalents) so i think it's safe to remove them for genres

https://github.com/sentriz/gonic/blob/master/server/ctrlsubsonic/handlers_by_tags.go#L157-L210

**Comment by Duncaen** [comment]
I removed the decoded field from the genre table, added a `getSongsByGenre` view and added genre parameter support to the `getRandomSongs` view.

One additional thing I could add is add a genre parameter to `spec.NewTrackByTags` so that all the views that return `TrackChild` can fill in the genre field.

**Comment by sentriz** [comment]
cool thank you :) i can have a proper look and merge when my next college work "sprint" ends. shouldn't be long

**Comment by Duncaen** [comment]
Ok nice, I rebased on master and squashed the fixup commits, this is ready for review.

File: server/ctrlsubsonic/handlers_common.go [hunk 1]
```diff
@@ -288,18 +288,27 @@ func (c *Controller) ServeGetSong(r *http.Request) *spec.Response {
 
 func (c *Controller) ServeGetRandomSongs(r *http.Request) *spec.Response {
```
Comment (sentriz) [inline server/ctrlsubsonic/handlers_common.go]: thanks for fixing my `year` laziness here :)

**Comment by sentriz** [comment]
sorry for the delay. nothing much i would change here. thank you for the feature :)

**Pull request merged**